### ISSUE TITLE
syncing eating some spaces between words

### ISSUE THREAD

**Issue by petRUShka** [baumgarr/nixnote2#295]
After syncing some spaces between words are missing. Seems like almost random but constant event.

For example: `около 16 часов` became `около 16часов` (Cyrillic letters).

**Comment by baumgarr** [comment]
How was rhe note created?

**Comment by artmg** [comment]
Other things that might help diagnose...
1) Do you edit the same notes between multiple platforms (e.g. Nixnote2, Android app, web browser)?
2) is this text that you typed in or pasted in?
3) Does this happen in single lines, multiple-line paragraphs, or when using any bullets or lists?
4) Does the space disappear on lines/paragraphs you edited, or elsewhere in the note?
Thanks

**Comment by petRUShka** [comment]
What does `rhe` mean? I don't remember it truly but I almost sure that note was created in NixNote2 

1. Yes it was Nixnote2 -> Web Browser -> Nixnote2
2. Typed and partially pasted in.
3. Only in bullets
4. Elsewhere. In som random places.

**Comment by artmg** [comment]
Hi @petRUShka, `rhe` must be a typo / autocorrect for `the`

I remember getting similar issues a year back, when I used to flip between Web / Android and Nixnote, so I have tried to reproduce the issue now, but with no success. Unfortunately I could NOT reproduce an issue with spaces being eaten (using nixnote2-daily+20170220). I do however have an issue where accented characters do not translate well between Web and Nixnote. I wonder if characters from the extended set are the cause? 

Would it be possible for you to paste here examples of the html code from **Show Source** of the bits of the notes where you get the issue, **before** and **after**. This will help us reproduce and work to a fix.
Thanks for your help

**Comment by baumgarr** [comment]
My guess is that it happened because of the fact it was initially copied from a web page.

Evernote has some strict requirements on the formatting of a note. Anything that deviates from this cannot sync with them. Because of this, you can run into issues when pasting from a web page or from any other HTML text document.

Part of the problem is that some HTML tags are just not permitted, and some of the attributes of other tags are not permitted. NixNote tries to do some cleanup and strip out tags that are not permitted just to enable it to sync. I'm guessing that whatever it stripped out caused the spaces to be lost. 

Sometimes you get lucky and clipping from a web page works, other times it doesn't. For clipping from a web page I recommend using Evernote's web clipper or (at the very least) trying some other extension that simplifies the web page. If you just need the text, you can also do a paste as unformatted text so none of the HTML tags are copied into a note.

@artmg, with regards to your accented character issues, please try to check "Force UTF8 Encoding" in the Debuging tab in the Preferences dialog. At some point in time Evernote stopped explicitly sending a tag to set this and NixNote was depending upon it. Right now it is disabled by default until I'm sure it doesn't create more problems than it solves.

**Comment by baumgarr** [comment]
No response. Closing issue.

**Comment by thompcinnamon** [comment]
So I actually have this issue regularly. I will edit or create notes in nixnote (2.0beta7), and I generally use bulleted lists. I will then open the note in Evernote's macOS client with missing spaces between words. The missing spaces almost seem to be in columns as if there is some line break in some internal version of the note that eats a space.

**Comment by baumgarr** [comment]
The first thing I'd try is to use a newer release. That one is pretty old. I don't recall if I fixed anything similar to this, but that has been a while ago.

Next question: Are you using non-latin characters?

**Comment by thompcinnamon** [comment]
That was the latest version in the repositories I was using. I've now installed 2.0. I haven't seen anything yet, but I'll check to be sure tomorrow when I sync to another computer, but so far it seems better viewing on evernote web.

I do not believe I was using any non-latin characters.

**Comment by thompcinnamon** [comment]
Interestingly, it works fine on the web interface, but it doesn't work properly on the Mac client. 

Therefore, there is a reasonable chance it's not an issue in NixNote2, though it is also true that it seems that this is only an issue with notes I've edited in NixNote2 (I don't know this for certain).

<img width="1552" alt="screen shot 2017-07-06 at 08 07 08" src="https://user-images.githubusercontent.com/8083606/27910302-8d8f1308-6222-11e7-92c9-47fdf1ad62c5.png">
<img width="1489" alt="screen shot 2017-07-06 at 08 07 27" src="https://user-images.githubusercontent.com/8083606/27910301-8d8dc8fe-6222-11e7-9287-cc971159d70a.png">

**Comment by thompcinnamon** [comment]
So I copied my test note and made a new one in NixNote2, then opened it in the Mac client, and exported it to get this: 
[test-note-2.enex.txt](https://github.com/baumgarr/nixnote2/files/1131823/test-note-2.enex.txt).

It seems that it is a line breaking issue. I'm not sure if it is how Evernote does it, or if it's something to do with NixNote2.

**Comment by sebiimaks** [comment]
I have this same problem and I'm using version 2.0. When I open notes made in NixNote2 on iOS or on a Mac, spaces are being eaten at regular intervals, but when I open the notes through the web client, there is no issue. Strange. Has anyone found a workaround for this?

**Comment by baumgarr** [comment]
I don't have an iOS machine, so debugging this could be difficult.

Can you provide me three copies of a note? 
1.) A note that was created on iOS and exported to an enex file without ever being used on NixNote.
2.) A copy of that same note created on NixNote before it was synchronized exported as an nnex file.
3.) A third copy of note #2, but after it was synchronized with the iOS client exported as an enex file from iOS.

My thought is that it is something different with the <en-note> attributes but since it works on the web client, I'm not exactly sure what.

Thanks.

**Comment by sebiimaks** [comment]
Thanks for the quick reply. I have attached the relevant notes. Probably ignore the actual _substance_ of the notes, as it's just gibberish. 

I can confirm that the space-eating problem occurs in the second paragraph when reading the note on an iOS device. Unfortunately, my Mac is in for repair and I can't find my Mac OS SSD for my x220, so I can't test it on the Mac client, although it reportedly produces the same errors there. 

Many thanks,
Sebastian


[NN Notes.zip](https://github.com/baumgarr/nixnote2/files/1202651/NN.Notes.zip)

**Comment by baumgarr** [comment]
I don't see anything obvious, but I have a couple of ideas. Do you have the ability to build NN from the source? If not, send me your email and if you are using the 32 or 64 bit version & I'll do a build for you.

**Comment by sebiimaks** [comment]
No, I have no idea how to do that. I'm using the 64bit version. My email is sebii####au@(googlemail).com

Just take out the hashes and change the latter to you know what.

**Comment by baumgarr** [comment]
I sent you a test build with some instructions. Let me know if it alters the behavior. Thanks.

**Comment by sebiimaks** [comment]
I was specifically syncing with the iOS client in step 3, but the problem exists in the Mac client as well, as I just had a look at my affected notes and they all have the same problem. I think the problem shows up with longer notes with multiple paragraphs in particular. The NN client is running with default settings, but I will check when I'm at home with my laptop.

I haven't tried replacing the binary you sent me because that's too high risk on that particular install. It has some work data that needs to be protected. I've been using the web client for notes of late, as I haven't been able to stop the issue from happening. It may well be Gnome or Manjaro that is forcing some kind of weird formatting onto it, but either way, I can't seem to stop it.

**Comment by rulameven** [comment]
This also started happening to me after updating to the latest Evernote client on my Windows 8 machine (6.7.5.5825 - CE Build ce-1.28.2184). I've been using Nixnote2 (v2.0) on Ubuntu and Evernote desktop client on Windows without any issues until the update. 

The notes are plain text, no bullets, multiple lines, only formatting used is italics, bold and tabs. The notes were created from plain text, by manual typing, no web clipping or copy-pasting involved. 

After editing in Nixnote2 and syncing, everything looks fine. After opening on Windows, random words get squashed together.

**Comment by rulameven** [comment]
Update: I downgraded the Evernote Windows client (to 6.7.4.5741) but it's still happening.

**Comment by rulameven** [comment]
Update 2: I've upgraded from Nixnote v2.0 to v2.0.2 and now can't reproduce the issue. I'm still using an older version of Evernote Windows client (6.7.5), just in case. I'll be very happy if I never encounter this issue again!
